contracts/AddressArrayLib.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity ^0.4.23;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Common : Address Array
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+
+library AddressArrayLib {
+
+    struct AddressArray {
+        address[] addresses;
+    }
+
+    //@dev Add element to array
+    function add(AddressArray storage array, address value) internal view returns (uint){
+        array.addresses.push(value);
+        return array.addresses.length;
+    }
+    //@dev Remove element from array by value
+    function removeByValue(AddressArray storage array, address value) internal view returns (uint) {
+        uint i = find(array, value);
+        removeByIndex(array, i);
```

Comment:
If the element is not found, then it still calls ``` removeByIndex(array, i); ```  and inside that method it does ``` array.addresses.length--; ``` even if the element was not found.

Need to add check for the i before calling ``` removeByIndex(array, i); ``` ?

Reply:
Or how exactly ``` array.addresses.length--;  ``` works ?

Reply:
In solidity, Dynamic arrays can be resized in storage (not in memory) by changing the .length property.
If element not found in the array, EVM will throw an exception and revert transaction.  However, I feel we should gracefully revert transaction explicitly.